ui/components/src/ECharts.tsx

```diff
@@ -0,0 +1,214 @@
+// Copyright 2021 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { useState, useLayoutEffect } from 'react';
+import debounce from 'lodash/debounce';
+import { ECharts, EChartsCoreOption, init } from 'echarts/core';
+import { Box, SxProps, Theme } from '@mui/material';
+
```

Comment:
ECharts does not export many of the internal types that we need, tracking [this issue](https://github.com/apache/echarts/issues/15284) and will hopefully be able to import most of these eventually